ambari-server/src/test/java/org/apache/ambari/server/api/services/mpackadvisor/commands/MpackAdvisorCommandTest.java

```diff
@@ -0,0 +1,291 @@
+
+    command.populateServiceAdvisors(objectNode);
+
+    // Verification for updated objectNode.
+    ArrayNode services = (ArrayNode) objectNode.get("services");
+    Iterator<JsonNode> servicesIter = services.getElements();
+    while (servicesIter.hasNext()) {
+      JsonNode service = servicesIter.next();
+      ObjectNode serviceVersion = (ObjectNode) service.get("StackServices");
+      if (serviceVersion.get("service_name").getTextValue().equals("HBASE")) {
+        assertEquals(serviceVersion.get("advisor_name").getTextValue(), advisorName);
+        assertEquals(serviceVersion.get("advisor_path").getTextValue(), advisorPath);
+      }
+    }
+  }
+
+  @Test
+  public void testGetServicesInformation() throws Exception {
+    final File recommendationsDir = temp.newFolder("recommendationDir");
+    String recommendationsArtifactsLifetime = "1w";
+    final int requestId = 2;
+    MpackAdvisorRunner maRunner = mock(MpackAdvisorRunner.class);
+    AmbariMetaInfo metaInfo = mock(AmbariMetaInfo.class);
+    doReturn(Collections.emptyList()).when(metaInfo).getStackParentVersions(anyString(), anyString());
+    final MpackAdvisorCommand<MpackAdvisorCommandTest.TestResource> command = spy(new MpackAdvisorCommandTest.TestMpackAdvisorCommand(
+        recommendationsDir, recommendationsArtifactsLifetime, ServiceInfo.ServiceAdvisorType.PYTHON, requestId, maRunner, metaInfo));
+
+    MpackAdvisorRequest request = MpackAdvisorRequest.MpackAdvisorRequestBuilder.forStack().build();
+
+    String hostsFilePath = Resources.getResource("mpack_advisor/hosts_count2.json").getPath();
+    String hostsJSON = new String(Files.readAllBytes(Paths.get(hostsFilePath)));
+
+    String servicesFilePath = Resources.getResource("mpack_advisor/ods_mpack_services.json").getPath();
+    servicesJSON = new String(Files.readAllBytes(Paths.get(servicesFilePath)));
+
+    File file = mock(File.class);
+    AmbariMetaInfo ambariMetaInfo = mock(AmbariMetaInfo.class);
+    MpackAdvisorCommand<MpackAdvisorCommandTest.TestResource> cmd = new MpackAdvisorCommandTest.TestMpackAdvisorCommand(
+        file, recommendationsArtifactsLifetime, ServiceInfo.ServiceAdvisorType.PYTHON, 1, maRunner, ambariMetaInfo);
+    String objectNodeFilePath = Resources.getResource("mpack_advisor/ods_mpack_objectnode.json").getPath();
+    String objectNodeStr = new String(Files.readAllBytes(Paths.get(objectNodeFilePath)));
+    ObjectNode objectNode = (ObjectNode) cmd.mapper.readTree(objectNodeStr);
+
+    MpackAdvisorCommand.MpackAdvisorData data = new MpackAdvisorCommand.MpackAdvisorData(hostsJSON, servicesJSON);
+    doReturn(hostsJSON).when(command).getHostsInformation(request);
+    doReturn(data).when(command).getServicesInformation(request, hostsJSON);
+    doReturn(objectNode).when(command).adjust(servicesJSON, request);
+
+    MpackAdvisorCommand.MpackAdvisorData result = command.getServicesInformation(request, hostsJSON);
+    assertEquals(result.servicesJSON, servicesJSON);
+  }
+
```

Comment:
Looks like it is the mock that is being tested, not the original code.

Reply:
Correct. Fixed it.